gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -40,50 +42,59 @@
   private static final String HADOOP_TOKEN_FILE_LOCATION = "HADOOP_TOKEN_FILE_LOCATION";
   private static final String MAPREDUCE_JOB_CREDENTIALS_BINARY = "mapreduce.job.credentials.binary";
 
-  private final Properties properties;
+  private final Closer closer = Closer.create();
   private final JobLauncher jobLauncher;
+  private final JobListener jobListener = new EmailNotificationJobListener();
 
   public AzkabanJobLauncher(String jobId, Properties props)
       throws Exception {
     super(jobId, LOG);
 
-    this.properties = new Properties();
-    this.properties.putAll(props);
+    Properties properties = new Properties();
```

Comment:
Azkaban and Voldemort both have their own Props class with strongly-typed
getter/setters and other sensical commodities.

Although copy-pasting is evil, it may be a lesser evil than using regular
Java Properties ;)

We should really just stick these things in a lib and re-use them wherever
needed...

-F
On Tue, May 5, 2015 at 18:12 sahilTakiar <notifications@github.com> wrote:

> In gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java
> <https://github.com/linkedin/gobblin/pull/120#discussion_r29728730>:
>
> >
> >    public AzkabanJobLauncher(String jobId, Properties props)
> >        throws Exception {
> >      super(jobId, LOG);
> >
> > -    this.properties = new Properties();
> > -    this.properties.putAll(props);
> > +    Properties properties = new Properties();
>
> Just some food for thought, but should we continue to rely on the Java
> Properties class. It has some weird behavior. Check out the Java Docs:
> http://docs.oracle.com/javase/7/docs/api/java/util/Properties.html
>
> Specifically the part about the put and putAll method. The methods
> directly modify the HashTable that backs the Properties object, and the
> methods allow a user to add non-String value types.
>
> I've also never been a fan of the new Properties(Properties defaults)
> constructor. It's confusing because it looks like a copy constructor, but
> in reality it sets property "defaults", which don't get returned by
> functions such as keySet().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/linkedin/gobblin/pull/120/files#r29728730>.
>